lib/spack/spack/spec.py

```diff
@@ -802,7 +802,7 @@ def _libs_default_handler(descriptor, spec, cls):
     Raises:
         RuntimeError: If no libraries are found
     """
-    name = 'lib' + spec.name
+    name = 'lib' + spec.name.replace('-', '?')
```

Comment:
@skosukhin  I forgot that @adamjstewart extended `find_libraries` to support globbing! Thanks for the explanation. As I said, I was missing something trivial.